documentation/External-Grafana/ReadMe.md

```diff
@@ -0,0 +1,179 @@
+
+## Out of scope
+
+In the design, integration to remote managed Grafana providers other than Amazon Managed Grafana is out of scope at this point. We may add additional integrations as part of future road map items.
+
+## Verification
+
+- Create integration tests for adding keys for remote Grafana instance.
+- Create integration tests to create Grafana dashboards on a remote Grafana Instance.
+- Create integration tests to add cloud data sources such as Amazon Managed Service for Prometheus, Amazon CloudWatch to remote Grafana instance.
+- Create integration tests to create alerting on a remote Grafana Instance.
+
+## Current
+
+Currently the grafana operator supports the following for only self managed Grafana Instance :
+- Adding remote data sources such as Amazon Managed Service for Prometheus.
+- Creating Dashboards.
+- Setting up alerting.
+- And Many more.
+
+## Proposal
+
+In short the proposal in this document is about enhancing the Grafana Operator to support the integration to managed grafana services. We would need to enhance the current version of Grafana Operator to support the following :
+
+- Defining external Grafana instance as Grafana source to Grafana operator.
+- Creating Grafana dashboards on external Grafana instance.
+- Adding data sources to external Grafana instance.
+- Creating alerting on external Grafana instance.
+
+### Defining external Grafana instance as Grafana source to Grafana operator.
+
+Today Grafana operator supports self managed grafana instance as a Grafana source to Grafana operator. With this new feature, we would like to add a new CRD by name `GrafanaInstance` which basically creates an identity to an external Grafana instance. 
+
+> CRD design of how a `GrafanaInstance` will look like. In this sample design, `url` should be preloaded in a ConfigMap and `grafana_api_key` should be loaded to a Secret. Choice of using external secrets or loading secrets manually is end user responsibility :
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaInstance
+metadata:
+  name: amazon-managed-grafana
+spec:
+  external:
+    url: <type ConfigMapSelector>
+    grafana_api_key: <type SecretKeySelector>
+```
+
+> CRD design for `GrafanaInstance` with a default setup for an external Grafana Instance. With this all Grafana Operator operations such as `GrafanaDataSource`, `GrafanaDashboard` will point to this default instance :
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaInstance
+metadata:
+  name: amazon-managed-grafana
+  annotations:
+    grafanainstance.kubernetes.io/is-default-grafana: "true"
+spec:
+  external:
+    url: <type ConfigMapSelector>
+    grafana_api_key: <type SecretKeySelector>
+```
+
+### Creating Grafana dashboards on Amazon Managed Grafana.
```

Comment:
This will be solved due to the label selectors that we already have in GrafanaDashboard so this can be removed.
We will solve this with the standard `Grafana` kind as mentioned earlier.